cpp/src/components/connectivity.cu

```diff
@@ -131,40 +146,85 @@ gdf_connected_components_impl(gdf_graph *graph,
     }
   else
     {
-      //dump error message and return unsupported, for now:
+      
+      //device memory requirements: 2n^2 + 2n x sizeof(IndexT) + 1 (for flag)
+      //( n = |V|)
       //
-      std::cerr<<"ERROR: Feature not supported, yet;"
-               <<" at: " << __FILE__ << ":" << __LINE__ << std::endl;
+      size_t n2 = 2*nrows;
+      n2 = n2*(nrows*sizeof(ByteT) + sizeof(IndexT)) + 1;
+
+      int device;
+      cudaDeviceProp prop;
+      
+      cudaGetDevice(&device);
+      cudaGetDeviceProperties(&prop, device);
+
+      if( n2 > prop.totalGlobalMem )
+        {
+
+          //not enough memory, dump error message and return unsupported:
+          //
+          std::cerr<<"ERROR: Insufficient device memory for SCC;"
+                   <<" at: " << __FILE__ << ":" << __LINE__ << std::endl;
+      
+          return GDF_MEMORYMANAGER_ERROR;
+        }
+      SCC_Data<ByteT, IndexT> sccd(nrows, p_d_row_offsets, p_d_col_ind);
+      size_t count = 0;
+      count = sccd.run_scc(p_d_labels);
       
-      return GDF_INVALID_API_CALL;//for now...
     }
+
+  //fill the vertex indices column:
+  //
+  thrust::sequence(thrust::device, p_d_verts, p_d_verts + nrows);
+  
   return GDF_SUCCESS;
 }
 
 /**
  * @brief Compute connected components. 
- * The weak version was imported from cuML.
+ * The weak version (for undirected graphs, only) was imported from cuML.
  * This implementation comes from [1] and solves component labeling problem in
  * parallel on CSR-indexes based upon the vertex degree and adjacency graph.
  *
  * [1] Hawick, K.A et al, 2010. "Parallel graph component labelling with GPUs and CUDA"
- * code is adapted / truncated from cuML: ml-prims/src/sparse/csr.h
+ * 
+ * The strong version (for directed or undirected graphs) is based on: 
+ * [2] Gilbert, J. et al, 2011. "Graph Algorithms in the Language of Linear Algebra"
+ *
+ * C = I | A | A^2 |...| A^k
+ * where matrix multiplication is via semi-ring: 
+ * (combine, reduce) == (&, |) (bitwise ops)
+ * Then: X = C & transpose(C); and finally, apply get_labels(X);
+ *
  *
- 
  * @param graph input graph; assumed undirected for weakly CC [in]
- * @param connectivity_type CUGRAPH_WEAK, CUGRAPH_STRONG  [in]
- * @param labels gdf_column for the output labels [out]
+ * @param connectivity_type CUGRAPH_WEAK or CUGRAPH_STRONG [in]
+ * @param table of 2 gdf_columns: output labels and vertex indices [out]
```

Comment:
Same here, we have the same comments describing this function in three different places (algorithms.h, gdf_connected_components, and gdf_connected_components_impl). I think this should appear only in the header file

Reply:
...and same answer as above.